packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -0,0 +1,340 @@
+      image = await layer.toImage(renderView.paintBounds, pixelRatio: 1.0);
+      return image.toByteData();
+    });
+
+    Future<Evaluation> evaluateNode(SemanticsNode node) async {
+      final SemanticsData data = node.getSemanticsData();
+      final List<SemanticsNode> children = <SemanticsNode>[];
+      Evaluation result = const Evaluation.pass();
+      node.visitChildren((SemanticsNode child) {
+        children.add(child);
+        return true;
+      });
+      for (SemanticsNode child in children)
+        result += await evaluateNode(child);
+      if (_shouldSkipNode(data))
+        return result;
+
+      // We need to look up the inherited text properties to determine the
+      // contrast ratio based on text size/weight.
+      double fontSize;
+      bool isBold;
+      final String text = (data.label?.isEmpty == true) ? data.value : data.label;
+      final List<Element> elements = find.text(text).evaluate().toList();
+      if (elements.length == 1) {
+        final Element element = elements.single;
+        final Widget widget = element.widget;
+        final DefaultTextStyle defaultTextStyle = DefaultTextStyle.of(element);
+        if (widget is Text) {
+          TextStyle effectiveTextStyle = widget.style;
+          if (widget.style == null || widget.style.inherit)
+            effectiveTextStyle = defaultTextStyle.style.merge(widget.style);
+          fontSize = effectiveTextStyle.fontSize;
+          isBold = effectiveTextStyle.fontWeight == FontWeight.bold;
+        } else if (widget is EditableText) {
+          isBold = widget.style.fontWeight == FontWeight.bold;
+          fontSize = widget.style.fontSize;
+        } else {
+          assert(false);
+        }
+      } else if (elements.length > 1) {
+        return const Evaluation.fail('Multiple nodes with the same label');
+      } else {
+        // If we can't find the text node, then look up the default text
+        fontSize = 12.0;
+        isBold = false;
+      }
+
+      // Transform local coordinate to screen coordinates.
+      Rect paintBounds = node.rect;
+      SemanticsNode current = node;
+      while (current != null) {
+        if (current.transform != null)
+          paintBounds = MatrixUtils.transformRect(current.transform, paintBounds);
+        paintBounds = paintBounds.shift(current.parent?.rect?.topLeft ?? Offset.zero);
+        current = current.parent;
+      }
+      final List<int> subset = _subsetToRect(byteData, paintBounds, image.width, image.height);
+      final _ContrastReport report = new _ContrastReport(subset);
+      final double contrastRatio = report.contrastRatio();
+      final double targetContrastRatio = (isBold && fontSize > kBoldTextMinimumSize) ?
+        kMinimumRatioLargeText : kMinimumRatioNormalText;
+      if (contrastRatio >= targetContrastRatio)
+        return result + const Evaluation.pass();
+      return result + new Evaluation.fail(
+        '$node:\nExpected contrast ratio of at least '
```

Comment:
Would it make sense to include a link to why these are the recommended contrasts in the reason? e.g. the links contained in the doc comments above?

(same for the failure reason of the other tester)

Reply:
Done